grammar.js

```diff
@@ -1045,14 +1071,10 @@ module.exports = grammar({
       ),
 
     list_body: ($) =>
-      general_body_rules(
-        'entry',
-        $.val_entry,
-        $._entry_separator,
-        $._newline,
-        null,
-        choice($._newline, punc().comma),
-      ),
+      general_body_rules('entry', $.val_entry, $._entry_separator, $._newline),
+
+    _list_body_or_empty: ($) =>
```

Comment:
going more off of https://github.com/nushell/tree-sitter-nu/pull/234#issuecomment-3582822457

`_list_body_or_empty` to me feels like a smell because such a type should not exist.

Ideally a list should have a simple rule:
`list: ($) => seq('[', repeat(choice($._list_entry, ',')), ']'),`

Handling newlines explicitly where their presence does not matter feels like a design problem (exclusion from `extras`).